tests/tizen_capi/unittest_tizen_capi.cc

```diff
@@ -5757,6 +5757,265 @@ TEST (nnstreamer_capi_singleshot, invoke_09_n)
 }
 #endif  /* ENABLE_ARMNN */
 
+/**
+ * @brief Test NNStreamer single shot (custom filter)
+ * @detail Run pipeline with custom filter with allocate in invoke, handle multi tensors.
+ */
+TEST (nnstreamer_capi_singleshot, invoke_10_p)
+{
+  const gchar cf_name[] = "libnnstreamer_customfilter_scaler_allocator" \
+      NNSTREAMER_SO_FILE_EXTENSION;
+  ml_single_h single;
+  ml_tensors_info_h in_info, out_info;
+  ml_tensors_data_h input, output;
+  ml_tensor_dimension in_dim;
+  int status;
+  unsigned int i;
+  void *data_ptr;
+  size_t data_size;
+
+  const gchar *root_path = g_getenv ("NNSTREAMER_SOURCE_ROOT_PATH");
+  gchar *test_model;
+
+  /* supposed to run test in build directory */
+  if (root_path == NULL)
+    root_path = "..";
+
+  test_model = g_build_filename (root_path, "build", "nnstreamer_example",
+      "custom_example_scaler", cf_name, NULL);
+  ASSERT_TRUE (g_file_test (test_model, G_FILE_TEST_EXISTS));
+
+  ml_tensors_info_create (&in_info);
+  ml_tensors_info_create (&out_info);
+
+  ml_tensors_info_set_count (in_info, 1);
+
+  in_dim[0] = 10;
+  in_dim[1] = 1;
+  in_dim[2] = 1;
+  in_dim[3] = 1;
+
+  ml_tensors_info_set_tensor_type (in_info, 0, ML_TENSOR_TYPE_INT16);
+  ml_tensors_info_set_tensor_dimension (in_info, 0, in_dim);
+
+  ml_tensors_info_clone (out_info, in_info);
+
+  status = ml_single_open (&single, test_model, in_info, out_info,
+      ML_NNFW_TYPE_CUSTOM_FILTER, ML_NNFW_HW_ANY);
+  ASSERT_EQ (status, ML_ERROR_NONE);
+
+  input = output = NULL;
+
+  /* generate input data */
+  status = ml_tensors_data_create (in_info, &input);
+  EXPECT_EQ (status, ML_ERROR_NONE);
+  ASSERT_TRUE (input != NULL);
+
+  for (i = 0; i < 10; i++) {
+    int16_t i16 = (int16_t) (i + 1);
+
+    status = ml_tensors_data_get_tensor_data (input, 0, &data_ptr, &data_size);
```

Comment:
Is your intent? 
If not, what about moving this code out of the loop?

Reply:
not my intent.
moved out.